idaes/apps/uncertainty_propagation/uncertainties.py

```diff
@@ -0,0 +1,371 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+
+import idaes
+import pandas as pd
+import numpy as np
+from scipy import sparse
+import pyomo.contrib.parmest.parmest as parmest
+from pyomo.environ import *
+from pyomo.opt import SolverFactory
+import shutil
+import logging
+
+logger = logging.getLogger('idaes.apps.uncertainty_propagation')
+
+def quantify_propagate_uncertainty(model_function, model_uncertain,  data, theta_names, obj_function=None, 
+                 tee=False, diagnostic_mode=False, solver_options=None):
+    """This function calculates error propagation of the objective function and constraints.  
```

Comment:
Please expand this docstring. What is `model_function` and `model_uncertain`? Why are they different? Do they need to be different?

Reply:
I expanded this docstring. The parmest uses `model_function` to estimate uncertain parameters. The uncertain parameters in  `model_uncertain` are fixed with the estimated values. The function `quantify_propagate_uncertainty` calculates error propagation of objective function and constraints in the `model_uncertain`.